securedrop/source_app/main.py

```diff
@@ -113,27 +120,40 @@ def submit():
                   "error")
             return redirect(url_for('main.lookup'))
 
-        fnames = []
         journalist_filename = g.source.journalist_filename
         first_submission = g.source.interaction_count == 0
 
         if msg:
             g.source.interaction_count += 1
-            fnames.append(
+            fname = (
                 store.save_message_submission(
                     g.filesystem_id,
                     g.source.interaction_count,
                     journalist_filename,
                     msg))
+            submission = Submission(g.source, fname)
+            db_session.add(submission)
+            db_session.commit()
+
         if fh:
-            g.source.interaction_count += 1
-            fnames.append(
-                store.save_file_submission(
-                    g.filesystem_id,
-                    g.source.interaction_count,
-                    journalist_filename,
-                    fh.filename,
-                    fh.stream))
+            for file_submitted in fh:
+                g.source.interaction_count += 1
+                fname = (
+                    store.save_file_submission(
+                        g.filesystem_id,
+                        g.source.interaction_count,
+                        journalist_filename,
+                        file_submitted.filename,
+                        file_submitted.stream))
+                submission = Submission(g.source, fname)
+                db_session.add(submission)
+                db_session.commit()
```

Comment:
Another note about this. We don't catch any errors, so if somethings goes bad (db failure, FS failure, etc), the source will get the generic 500 error page. I think there are two solutions.

1. Keep going, collect all errors, and display all errors at the end.
2. Halt immediately, display which file erred and which files were not processed.